lib/Common/PlatformAgnostic/Arrays.h

```diff
@@ -0,0 +1,40 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#pragma once
+
+namespace PlatformAgnostic
+{
+namespace Arrays
+{
+#ifndef ENABLE_GLOBALIZATION
+
+    class ArrayLocalization
+    {
+        char16 localeSeparator;
+
+        static const uint16 BufSize = 256;
+
+    public:
+
+        ArrayLocalization();
+
+        inline char16 GetLocaleSeparator() { return localeSeparator; }
+    };
+
+#endif   
+
+    static const uint32 SeparatorBufferSize = 6;
+
+    // According to MSDN the maximum number of characters for the list separator (LOCALE_SLIST) is four, including a terminating null character.
```

Comment:
Just to add, the LOCALE_SLIST lctype is sort of somewhat deprecated, even though nothing is mentioned on MSDN. There is only one usage of it (Office Excel) AFAIK, and there is *no* mapping for it to any CLDR data points.
CLDR has a much richer notion of lists, and there is no one single simple data point that directly corresponds to it.
Also, there is no ICU equivalent either.
Thus, I'm not exactly sure what the longer term plan is for LOCALE_SLIST.

Reply:
Looks like in ICU-land, we would use icu::ListFormatter.... if we had access to the C++ API. I can't find a C version exposed. I don't think its urgent, but that is a possible gap to address in the distant future.

Reply:
Oh don't get me wrong, I'm not saying that this pull-request is bad in any way (thank you very kindly for your contribution! 👍 )
I just wanted to point out that the LOCALE_SLIST is somewhat problematic, and might need some more thinking in the future.

As for the C vs C++ API support on the ICU side, if there becomes a need for this (if it gets added to the spec) then I would hope the new plan of record from ECMA/TC-39 of needing C APIs would help in that regard.

Reply:
Technically, there is an Intl version of Array.prototype.toLocaleString, but as always, the entire spec is best-effort and as long as Chrome and V8 aren't trying any harder than we are, I don't see a reason to worry about it.